bin/reth/src/dirs.rs

```diff
@@ -28,67 +28,103 @@ pub fn config_dir() -> Option<PathBuf> {
     dirs_next::config_dir().map(|root| root.join("reth"))
 }
 
-/// A wrapper type that either parses a user-given path for the reth database or defaults to an
-/// OS-specific path.
-#[derive(Clone, Debug)]
-pub struct DbPath(PathBuf);
+/// Returns the path to the reth cache directory.
+///
+/// Refer to [dirs_next::cache_dir] for cross-platform behavior.
+pub fn cache_dir() -> Option<PathBuf> {
+    dirs_next::cache_dir().map(|root| root.join("reth"))
+}
 
-impl Display for DbPath {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0.display())
-    }
+/// Returns the path to the reth logs directory.
+///
+/// Refer to [dirs_next::cache_dir] for cross-platform behavior.
+pub fn logs_dir() -> Option<PathBuf> {
+    cache_dir().map(|root| root.join("logs"))
 }
 
-impl Default for DbPath {
-    fn default() -> Self {
-        Self(database_path().expect("Could not determine default database path. Set one manually."))
+/// Returns the path to the reth database.
+///
+/// Refer to [dirs_next::data_dir] for cross-platform behavior.
+#[derive(Default, Debug, Clone)]
+pub struct DbPath;
+
+impl XdgPath for DbPath {
+    fn resolve() -> Option<PathBuf> {
+        database_path()
     }
 }
 
-impl FromStr for DbPath {
-    type Err = shellexpand::LookupError<VarError>;
+/// Returns the path to the default reth configuration file.
+///
+/// Refer to [dirs_next::config_dir] for cross-platform behavior.
+#[derive(Default, Debug, Clone)]
+pub struct ConfigPath;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        Ok(Self(parse_path(s)?))
+impl XdgPath for ConfigPath {
+    fn resolve() -> Option<PathBuf> {
+        config_dir().map(|p| p.join("reth.toml"))
     }
 }
 
-impl AsRef<Path> for DbPath {
-    fn as_ref(&self) -> &Path {
-        self.0.as_path()
+/// Returns the path to the reth logs directory.
+///
+/// Refer to [dirs_next::cache_dir] for cross-platform behavior.
+#[derive(Default, Debug, Clone)]
+pub struct LogsDir;
+
+impl XdgPath for LogsDir {
+    fn resolve() -> Option<PathBuf> {
+        logs_dir()
     }
 }
 
-/// A wrapper type that either parses a user-given path for the reth config or defaults to an
+/// A small helper trait for unit structs that represent a standard path following the XDG
+/// path specification.
+trait XdgPath {
+    fn resolve() -> Option<PathBuf>;
+}
+
+/// A wrapper type that either parses a user-given path or defaults to an
 /// OS-specific path.
+///
+/// The [FromStr] implementation supports shell expansions and common patterns such as `~` for the
+/// home directory.
+///
+/// # Example
+///
+/// ```rs
+/// let default: StandardPath<DbPath> = StandardPath::default();
+/// let custom = StandardPath::from_str("my/path/to/db").unwrap();
+///
+/// println!("Default db path: {default}, custom db path: {custom}");
+/// ```
 #[derive(Clone, Debug)]
-pub struct ConfigPath(PathBuf);
+pub struct StandardPath<D>(PathBuf, std::marker::PhantomData<D>);
```

Comment:
this seems useful,
but should we perhaps rename this to convey that this is

1. Os Specific behavior
2. Expandable

`OsExpandablePath` or something?